juju/juju
# [JUJU-4550] Move cloud credentials from mongo to dqlite

There's a lot of mechanical changes to weave the credential service into the api facades The cloud facade has a lot of churn, as does the relevant (old) state logic

The facades and the state policy instance need to access a model's credential. They used to be able to do this just by using the usual state.State. However, now the credential must be access via the new domain/credential service. So this service needs to be made available to the facades that need it and also the state policy constructor.

Many of the changes are simply to the facades Register() functions, to pass in the credential service obtained from the context.

To make the credential service available to the state policy, the state worker manifold is updated to have the service factory as an input, and the credential service is obtained when the state pool is created.
The state policy uses the credential to make an environ, which is used when validating deployments etc, so the various helper methods to set this  were also updated to use the credential service.

The other major change is for code that deals with managing model credentials, eg marking a model as suspended when the credential is invalid, or loading status info about a model and/or credential. The credential code is lifted out of state; what remains is just the credential tag attribute on the model.

On k8s, dqlite watchers are supported yet, and so the workers which use them bounce. This kills the credential watcher which stops juju from working. For now a noop stub is added to allowed k8s controllers to at least function.

## QA steps

bootstrap
add model
deploy an application

juju show-model
```
...
  credential:
    name: juju
    owner: admin
    cloud: aws
    validity-check: valid
```

juju show-credential aws juju
```
controller-credentials:
  aws:
    juju:
      content:
        auth-type: access-key
        validity-check: valid
        access-key: xxxx
      models:
        controller: admin
```
